azkaban-common/src/main/java/azkaban/utils/SystemMemoryInfo.java

```diff
@@ -55,9 +55,9 @@ public static void init(int memCheckInterval) {
    * Given Xms/Xmx values (in kb) used by java process, determine if system can
    * satisfy the memory request
    */
-  public synchronized static boolean canSystemGrantMemory(long xms, long xmx, long freeMemDecrAmt) {
+  public synchronized static boolean canSystemGrantMemory(long xms, long xmx, long freeMemDecrAmt) throws Exception {
     if (!memCheckEnabled) {
-      return true;
+      throw new Exception("memory check is disabled");
```

Comment:
this method returning true should only indicate memory is granted on request. Since we rely on the return value of this method to construct the retry logic.  If if memory check is not enabled,  it should throws an exception.